File: src/data_structures/stack/Stack.kt
package data_structures.stack

class Stack<T>() {
    companion object {
        fun <T> create(list: Iterable<T>): Stack<T> {
            val stack = Stack<T>()
            list.forEach {
                stack.push(it)
            }
            return stack
        }

        fun <T> stackOf(vararg item: T): Stack<T> {
            return create(item.asList())
        }
    }

    private val storage = arrayListOf<T>()

    val size: Int
        get() = storage.size

    val isEmpty: Boolean
        get() = size == 0

    fun push(element: T) {
        storage.add(element)
    }

    fun pop(): T? {
        if (!isEmpty) {
            return storage.removeAt(storage.size - 1)
        }
        return null
    }


    override fun toString(): String = buildString {
        appendln("=====top=====")
        storage.asReversed().forEach {
            appendln("$it")
        }
        appendln("=============")
    }
}

File: data_structures/stack/stack.py
class Stack:
    def __init__(self):
        self.list = list()

    @property
    def is_empty(self):
        return len(self.list) == 0

    def push(self, element):
        self.list.insert(0,element)
        return self

    def pop(self):
        if self.is_empty:
            return None
        return self.list.pop()

    def to_string(self):
        str = "----top---\n"
        for i in self.list[::-1]:
            str += '{0}\n'.format(i)
        str += '----bottom----'
        return str
